omnipaxos_core/src/util.rs

```diff
@@ -336,3 +342,29 @@ impl SequenceNumber {
         }
     }
 }
+
+
+pub(crate) struct LogicalClock {
+    time: u64,
+    timeout: u64,
+}
+
+impl LogicalClock {
+    pub fn with(timeout: u64) -> Self {
+        Self {
+            time: 0,
+            timeout
+        }
+    }
+
+    pub fn tick(&mut self) {
+        self.time += 1;
+        if self.time == self.timeout {
+            self.time = 0;
+        }
+    }
+
+    pub fn reached_timeout(&self) -> bool {
+        self.time == 0
+    }
```

Comment:
Since we always check the timeout after each tick, I think a better API is to have one function `tick_and_check_timeout() -> bool`. This way the rust compiler will also complain if we call it and disregard the result (which will help us avoiding mistakes when coding).